sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveSessionCatalog.scala

```diff
@@ -69,7 +69,11 @@ private[sql] class HiveSessionCatalog(
     // Current thread context classloader may not be the one loaded the class. Need to switch
     // context classloader to initialize instance properly.
     Utils.withContextClassLoader(clazz.getClassLoader) {
-      Try(super.makeFunctionExpression(name, clazz, input)).getOrElse {
+      val t = Try(super.makeFunctionExpression(name, clazz, input))
```

Comment:
I think we should fail earlier if `super.makeFunctionExpression` fails due to invalid inputs.

Reply:
> I think we should fail earlier if `super.makeFunctionExpression` fails due to invalid inputs.

Got you point， if throw exception about Invalid input means it's a Spark UDF, and this place only throw this type exception  when match but failed.

Reply:
> I think we should fail earlier if `super.makeFunctionExpression` fails due to invalid inputs.

When throw `No handler for UDF/UDAF/UDTF` mean both spark and hive UDF definition are not matched

Reply:
> I think we should fail earlier if `super.makeFunctionExpression` fails due to invalid inputs.

@cloud-fan How about  current change.